src/mbgl/map/backend_scope.cpp

```diff
@@ -0,0 +1,26 @@
+#include <mbgl/map/backend_scope.hpp>
+#include <mbgl/map/backend.hpp>
+#include <mbgl/util/thread_local.hpp>
+
+namespace mbgl {
+
+static util::ThreadLocal<Backend> currentBackend;
+
+BackendScope::BackendScope(Backend& backend_)
+    : priorBackend(currentBackend.get()),
+      backend(backend_) {
+    backend.activate();
+    currentBackend.set(&backend);
```

Comment:
Can we check whether the backend is identical to the previously enabled backend and make activation/deactivation a no-op in this case?